src/hotspot/share/interpreter/bytecodeStream.hpp

```diff
@@ -136,9 +136,10 @@ class RawBytecodeStream: public BaseBytecodeStream {
       assert(code != Bytecodes::_wide && code != Bytecodes::_tableswitch
              && code != Bytecodes::_lookupswitch, "can't be special bytecode");
       _is_wide = false;
-      _next_bci += len;
-      if (_next_bci <= _bci) { // Check for integer overflow
+      if (INT_MAX - len <= _next_bci) { // Check for integer overflow
```

Comment:
I don't see how we can ever hit this case, given the `_bci <= _end_bci - len` test above.

Reply:
Hm, that seems right. Then the integer overflow check never did anything at all, anyway, right?

Reply:
It looks that way.  In a product build it seems to get completely optimized out.  Unless I'm missing something, it should be safe to remove.

Reply:
Good, I agree. Let's remove it.